04-complexity-introduction/intro_complexity_answers.txt

```diff
@@ -0,0 +1,87 @@
+1. Describe an analogy for relating an algorithm that has efficiency O(1) and another algorithm that has O(2^n). An example would be:
+  An algorithm that has O(1) is a Cheetah and an algorithm that has O(2^n) is a Snail.
+
+    An algorithm that has O(1) takes constant time to execute no matter how big the input size is. An algorithm that has O(2^n) takes exponential time to execute even with a small size of input. Thus the former algorithm is like the latest model of CPU and the latter algorithm is like the first computer 'ENIAC'.
+
+
+
+2. In plain English, what is the best case scenario for binary search?
+
+    The target item is located in the middle of the collection.
+
+
+
+3. In plain English, what is the worst-case scenario for binary search?
+
+    The target item is located in the beginning of the collection or at the end of the collection.
+
+
+4. In plain English, what is the bounded-case scenario for binary search?
+
+    The target item is located between the middle item and the leftmost/rightmost item.
```

Comment:
The bounded-case scenario is the performance between the best-case and the worst-case. So when an element that the method is looking at isn't at the middle of collection nor at the leftmost-rightmost index, that is the bounded-case.